Dokan/Dokan/Scenes/ReviewProduct/Views/RatingProgressDetails/RatingProgressDetails.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  RatingProgressDetails.swift
+//  Dokan
+//
+//  Created by marwa on 26/08/2022.
+//
+
+import UIDokan
+import UIKit
+
+class RatingProgressDetails: UIView {
+
+    // MARK: - Outlets
+
+    @IBOutlet private(set) weak var contentView: UIView!
+    @IBOutlet private(set) weak var starRating: StarsView!
+    @IBOutlet private(set) weak var progressBarRating: UIProgressView!
+    @IBOutlet private(set) weak var starsRatingLabel: UILabel!
+
+    // MARK: - Properties
+
+    var progressDetails: RatingProgressDetailsViewModel? {
+        didSet {
+            guard let progressDetails = progressDetails else {
+                return
+            }
+
+            starRating.applyStyleToView(Double(progressDetails.starNumber))
+            starsRatingLabel.text = String(progressDetails.rating)
+            updateProgressBarRating(progressDetails: progressDetails)
+        }
+    }
+
+    // MARK: - initializer
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        viewSetup()
+    }
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        viewSetup()
+    }
+}
```

Comment:
Read about it... Much easier and lesser code
```suggestion
class RatingProgressDetails: UIViewFromNib {

    // MARK: - Outlets

    @IBOutlet private(set) weak var contentView: UIView!
    @IBOutlet private(set) weak var starRating: StarsView!
    @IBOutlet private(set) weak var progressBarRating: UIProgressView!
    @IBOutlet private(set) weak var starsRatingLabel: UILabel!

    // MARK: - Properties

    var progressDetails: RatingProgressDetailsViewModel? {
        didSet {
            guard let progressDetails = progressDetails else {
                return
            }

            starRating.applyStyleToView(Double(progressDetails.starNumber))
            starsRatingLabel.text = String(progressDetails.rating)
            updateProgressBarRating(progressDetails: progressDetails)
        }
    }
}
```